lib/driver.js

```diff
@@ -484,6 +484,33 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  async startMjpegStream() {
+      // turn on mjpeg stream reading if requested
+      if (this.opts.mjpegScreenshotUrl) {
+        this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+        this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+        await this.mjpegStream.start();
+      }
+  }
+
+  async allocateMjpegServerPort(caps) {
+    this.log.debug(
+      `MJPEG broadcasting requested, forwarding MJPEG server port ${this.opts.mjpegServerPort} ` +
+      `to local port ${this.opts.mjpegServerPort}`
+    );
+    try {
+      await DEVICE_CONNECTIONS_FACTORY.requestConnection(caps.udid, this.opts.mjpegServerPort, {
+        devicePort: this.opts.mjpegServerPort,
```

Comment:
I am not sure this is going to work. The device only exposes MJPEG streams on a single port 9100. A local port number might vary though